### PR TITLE
MES-2908 CPC Detailed Test View

### PR DESCRIPTION
**Pull request by tomcrawleyy**
## Description
Introduces logic to allow CPC tests to be viewed from 'Search for a completed test'
Introduces new components which are CPC specific.
## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [x] Squashed commit contains the JIRA ticket number

## Screenshots (optional)
Cat CCPC Fail Vehicle Details:
Screenshot 2020-06-30 at 10 59 50

Cat CCPC Fail Debrief:
Screenshot 2020-06-30 at 11 00 09

Cat CCPC Test Summary Fail:
Screenshot 2020-06-30 at 13 09 54

Cat CCPC Pass Vehicle Details:
Screenshot 2020-06-30 at 10 52 57

Cat CCPC Pass Debrief:
Screenshot 2020-06-30 at 10 53 27

Cat DCPC Vehicle Details Hidden (usually situated between these components):
Screenshot 2020-06-30 at 13 21 17

### CODE REVIEW

File: src/components/common/cpc-debrief-card/cpc-debrief-card.ts [hunk 1]
```diff
@@ -52,11 +61,14 @@ export class CPCDebriefCardComponent implements OnInit {
...
 
   isPass = (): boolean => {
-    return this.testOutcome === TestOutcome.PASS;
+    return this.testOutcome === TestOutcome.PASS || this.testOutcome === OutcomeType.Passed;
   }
 
   ngOnInit(): void {
     this.questions = [this.question1, this.question2, this.question3, this.question4, this.question5];
+    if (this.isDetailedTestView) {
+      this.cardHeader = 'Debrief';
```
Comment (Cullima) [inline src/components/common/cpc-debrief-card/cpc-debrief-card.ts]: is this heading change in the ACs?

**Review by Cullima — CHANGES_REQUESTED** [review]
Looks good just a small question around changing the card header. Can you add
- description
- pass, fail and terminate screen captures
- unit tests

File: src/components/common/cpc-debrief-card/cpc-debrief-card.ts (hunk 1 above)
Comment (tomcrawleyy) [reply src/components/common/cpc-debrief-card/cpc-debrief-card.ts]: It displays as 'Debrief' in the screenshot that Lewis attached which is what I was referencing when designing this component, but no its not specifically mentioned in Lukes' ACs, just seemed wrong to keep it as 'Test details' when it specifically says 'Debrief' in the attachment, this doesnt affect anything outside of the detailed test view

Comment (tomcrawleyy) [reply src/components/common/cpc-debrief-card/cpc-debrief-card.ts]: just confirmed with Luke - this is how they want it

**Comment by andrewsetterfield** [comment]
Test Details
	Business name - not present
	Entitlement check - not present

Business details - not present

Order of sections doesn’t match design/details in story.

For the not present fields can you confirm with Luke where they're to be sourced from or have him remove them.

**Comment by andrewsetterfield** [comment]
The close button on the Test information screen looks wrong.  
Adi
Screenshot 2020-06-25 at 13 51 10
CCPC
Screenshot 2020-06-25 at 13 51 36

**Comment by tomcrawleyy** [comment]
> Test Details
> Business name - not present
> Entitlement check - not present
> 
> Business details - not present
> 
> Order of sections doesn’t match design/details in story.
> 
> For the not present fields can you confirm with Luke where they're to be sourced from or have him remove them.

theres nothing business related in the schema, think thats why i removed it, ill check with entitlement tho

**Comment by andrewsetterfield** [comment]
The combination is ment to be inline with the other text, but its over by the percentages.

Screenshot 2020-06-25 at 13 56 14

**Comment by andrewsetterfield** [comment]
Test summary section isnt correct.  extra fields present and some missing

File: src/shared/mocks/cat-cpc-test-result.mock.ts [hunk 2]
```diff
@@ -0,0 +1,242 @@
+import { TestResultCatCPCSchema } from '@dvsa/mes-test-schema/categories/CPC';
+
+export const categoryCPCTestResultMock: TestResultCatCPCSchema = {
+  version: '0.0.1',
+  category: 'F',
```
Comment (matthew2564) [inline src/shared/mocks/cat-cpc-test-result.mock.ts]: update category to be either CCPC or DCPC

File: src/pages/view-test-result/cat-cpc/components/test-summary-card/test-summary-card.html [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  <ion-card-content>
+    <ion-grid>
+      <data-row [label]="'Accompanied by'" [value]="getAccompaniedBy()"></data-row>
+      <data-row
+        [label]="'Licence provided'"
+        [value]="getProvisionalLicenceProvided()"
+        *ngIf="shouldDisplayLicenceProvided(getProvisionalLicenceProvided())">
+      </data-row>
+      <data-row
+        [label]="'Code 78'"
```
Comment (andrewsetterfield) [inline src/pages/view-test-result/cat-cpc/components/test-summary-card/test-summary-card.html]: is this relevant to CPC

File: src/pages/view-test-result/cat-cpc/__tests__/view-test-result.cat-cpc.page.spec.ts [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+      });
+    });
+    describe('getTestDetails', () => {
+      it('should correctly generate the data', () => {
+        component.testResult = categoryCPCTestResultMock;
+
+        const result: TestDetailsModel = component.getTestDetails();
+
+        expect(result.applicationReference).toBe('12345672013');
+        expect(result.category).toBe('F');
```
Comment (matthew2564) [inline src/pages/view-test-result/cat-cpc/__tests__/view-test-result.cat-cpc.page.spec.ts]: please update this to be CCPC/DCPC when mock data is updated

**Review by andrewsetterfield — APPROVED** [review]

**Review by Cullima — APPROVED** [review]

**Pull request merged**